compiler/noirc_driver/src/lib.rs

```diff
@@ -332,15 +340,15 @@ pub fn compile_no_check(
     if !(force_compile || options.print_acir || options.show_brillig || options.show_ssa) {
         if let Some(cached_program) = cached_program {
             if hash == cached_program.hash {
-                return Ok(cached_program);
+                return Ok((cached_program, Vec::new()));
```

Comment:
We cache errors already I believe, so I think we may need to do the same for warnings

Reply:
I would not expect that we cache errors, if there is an error, we cannot output the program so we would not cache it anyways, would we?

Reply:
Caching a program and running from the cache should have the same behavior as if we were running the program twice. If warnings aren't displayed the second time, that'd break that property

Reply:
Yes I agree, but I don't think we cache failing programs, i.e errors. I will look for caching warnings.

Reply:
We do - there was a recent issue where running the cached version of a program would change the location on the error message. Here's what I get when I run this PR twice on the code:
```rs
fn main() -> pub Field {
    assert(false);
    3
}
```
output:
```
?pr/3205 ~/Code/Noir/noir/short> nargo execute
warning: Returning a constant value is not allowed
  ┌─ /../noir/short/src/main.nr:4:5
  │
4 │     3
  │     - constant value
  │
  = Call stack:
    1. /../noir/short/src/main.nr:4:5

error: Failed constraint
  ┌─ /../noir/short/src/main.nr:3:12
  │
3 │     assert(false);
  │            -----
  │
  = Call stack:
    1. /../noir/short/src/main.nr:3:12

Cannot satisfy constraint
exit 1
?pr/3205 ~/Code/Noir/noir/short> nargo execute
error: Failed constraint
  ┌─ /../noir/short/src/main.nr:3:12
  │
3 │     assert(false);
  │            -----
  │
  = Call stack:
    1. /../noir/short/src/main.nr:3:12

Cannot satisfy constraint
exit 1
```

Reply:
I am closing the conversation because I created an issue for this.